### PR TITLE
Adding two new operations for Reservations

### PR DESCRIPTION
**Pull request by kc786**
1. Reservation Summaries
2. Reservation Details

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json [hunk 1]
```diff
@@ -117,6 +117,127 @@
...
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReservationSummariesListResult"
```
Comment (veronicagg) [inline specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: would you want to add x-ms-pageable to this operation- https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-pageable 
You can set nextLinkName to null, which results in an enumerable for some of the SDKs.

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json [hunk 2]
```diff
@@ -117,6 +117,127 @@
...
+            "name": "scope",
+            "description": "The scope of the reservation summaries. The scope can be 'providers/Microsoft.Capacity/reservationorders/{ReservationOrderId}' or 'providers/Microsoft.Capacity/reservationorders/{ReservationOrderId}/reservations/{ReservationId}'",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-skip-url-encoding": true
+          },  
+          {
+            "name": "grain",
+            "description": "Can be daily or monthly",
```
Comment (veronicagg) [inline specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: should it be an enum?

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json [hunk 3]
```diff
@@ -117,6 +117,127 @@
...
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReservationDetailsListResult"
```
Comment (veronicagg) [inline specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: same comment as previous, would you want to mark this as x-ms-pageable?

**Review by veronicagg — CHANGES_REQUESTED** [review]
Left a couple of comments inline. Rest looks good from my side. 
Adding @ravbhatnagar for ARM sign-off as PR adds 2 operations to the spec.

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json (hunk 2 above)
Comment (kc786) [reply specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: pushed my changes

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json (hunk 1 above)
Comment (kc786) [reply specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: pushed my changes

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json (hunk 3 above)
Comment (kc786) [reply specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: pushed my changes

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json [hunk 4]
```diff
@@ -155,7 +155,8 @@
             "description": "Can be daily or monthly",
             "in": "query",
             "required": true,
-            "type": "string"
+            "type": "string",
+            "enum": ["daily", "monthly"]
```
Comment (veronicagg) [inline specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: please take a look at adding https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum and most likely modelAsString=true

**Review by veronicagg — CHANGES_REQUESTED** [review]
left one more comment, rest looks good from my side. 
Waiting on @ravbhatnagar for approval.

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json (hunk 4 above)
Comment (kc786) [reply specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: pushed the change

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json [hunk 5]
```diff
@@ -156,7 +156,8 @@
             "in": "query",
             "required": true,
             "type": "string",
-            "enum": ["daily", "monthly"]
+            "enum": ["daily", "monthly"],
+            "modelAsString":true
```
Comment (veronicagg) [inline specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: please take a look at https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum
there's a specific format to follow to add the x-ms-enum extension.

**Review by veronicagg — CHANGES_REQUESTED** [review]
left comment inline = please review format for https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json (hunk 5 above)
Comment (kc786) [reply specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: Done

**Review by kc786 — COMMENTED** [review]
corrected the format to add  x-ms-enum extension

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json [hunk 6]
```diff
@@ -157,7 +157,21 @@
             "required": true,
             "type": "string",
             "enum": ["daily", "monthly"],
-            "modelAsString":true
+            "x-ms-enum": {
+              "name": "datagrain",
+              "modelAsString":true,
+              "values":[
+                        {
```
Comment (veronicagg) [inline specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: minor: formatting/indenting seems off, if you could use 2 spaces for formatting it'd be great.

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json [hunk 7]
```diff
@@ -157,7 +157,21 @@
...
+            "x-ms-enum": {
+              "name": "datagrain",
+              "modelAsString":true,
+              "values":[
+                        { 
+                          "value":"daily",
+                          "description":"Daily grain of data",
+                          "name":"DailyGrain"                          
+                        },
+                        { "value":"monthly",
```
Comment (veronicagg) [inline specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: minor: if you could be consistent with formatting here, and adding a new line after the "{", it'd be great.

**Review by veronicagg — CHANGES_REQUESTED** [review]

**Comment by ravbhatnagar** [comment]
Looks good. Just camelcase the resource type names - reservationDetails and reservationSummaries

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json (hunk 6 above)
Comment (kc786) [reply specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: Done

File: specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json (hunk 7 above)
Comment (kc786) [reply specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json]: Done

**Review by veronicagg — APPROVED** [review]
Thanks @kc786 and @ravbhatnagar . Looks good to me!

**Pull request merged**